avocado/plugins/runner_nrunner.py

```diff
@@ -86,66 +164,88 @@ def _get_all_runtime_tasks(self, test_suite):
-            }
-            job.result.start_test(early_state)
-            job.result_events_dispatcher.map_method('start_test',
-                                                    job.result,
-                                                    early_state)
-
-            task.status_services = []
-            for status in task.run():
-                status_repo.process_message(status)
-                result_dispatcher.map_method('test_progress', False)
-
-                if status['status'] not in ["started", "running"]:
-                    break
-
-            # test execution time is currently missing
-            # since 358e800e81 all runners all produce the result in a key called
-            # 'result', instead of 'status'.  But the Avocado result plugins rely
-            # on the current runner approach
-            this_task_data = status_repo.get_task_data(task.identifier)
-            test_state = {'status': this_task_data[-1]['result'].upper()}
-            test_state.update(early_state)
-
-            time_start = this_task_data[0]['time']
-            time_end = this_task_data[-1]['time']
-            time_elapsed = time_end - time_start
-            test_state['time_start'] = time_start
-            test_state['time_end'] = time_end
-            test_state['time_elapsed'] = time_elapsed
-
-            # fake log dir, needed by some result plugins such as HTML
-            test_state['logdir'] = ''
-
-            # Populate task dir
-            base_path = os.path.join(job.logdir, 'test-results')
-            self._populate_task_logdir(base_path,
-                                       task,
-                                       this_task_data,
-                                       job.config.get('core.debug'))
-
-            job.result.check_test(test_state)
-            result_dispatcher.map_method('end_test', job.result, test_state)
+
+        status_server_uri = test_suite.config.get('nrunner.status_server_uri')
+        self._start_status_server(status_server_uri)
+
+        # pylint: disable=W0201
+        self.tasks = self._get_all_runtime_tasks(test_suite)
+        if test_suite.config.get('nrunner.shuffle'):
+            random.shuffle(self.tasks)
+        tsm = TaskStateMachine(self.tasks)
+        spawner_name = test_suite.config.get('nrunner.spawner')
+        spawner = SpawnerDispatcher()[spawner_name].obj
+        max_running = test_suite.config.get('nrunner.max_parallel_tasks')
+        workers = [Worker(tsm, spawner, max_running=max_running).run()
+                   for _ in range(max_running)]
+        asyncio.ensure_future(self._update_status(job))
+        loop = asyncio.get_event_loop()
+        try:
+            loop.run_until_complete(asyncio.wait_for(asyncio.gather(*workers),
+                                                     job.timeout or None))
+        except (KeyboardInterrupt, asyncio.TimeoutError):
+            summary.add("INTERRUPTED")
+
```

Comment:
Are logging messages possible in this async situation? Just a thought but I think it is much easier if there is more logging information (e.g. `LOG_JOB.debug`) if we develop further functionality on top of this logic.

Reply:
Yes, there's no limitation I'm aware of.  it should be possible to do more logging.